### PR TITLE
Out-of-bounds while get port from host field

### PR DESCRIPTION
**Pull request by oknet**

### CODE REVIEW

**Comment by oknet** [comment]
[approve ci autest]

**Comment by scw00** [comment]
Could we use the new literal (like `port_prt_len`) and change the function to `get_host_port_values(nullptr, &m_host_length, &port_ptr, &port_prt_len)`.

File: proxy/hdrs/HTTP.cc [hunk 1]
```diff
@@ -1558,7 +1558,7 @@ HTTPHdr::_fill_target_cache() const
              (m_host_mime = const_cast<HTTPHdr *>(this)->get_host_port_values(nullptr, &m_host_length, &port_ptr, nullptr))) {
```
Comment (scw00) [inline proxy/hdrs/HTTP.cc]: Could we use the new literal (like port_prt_len) and change the function to get_host_port_values(nullptr, &m_host_length, &port_ptr, &port_prt_len) ?

Comment (oknet) [reply proxy/hdrs/HTTP.cc]: get port_len from ```get_host_port_values()```

**Comment by scw00** [comment]
[approve ci autest]

**Comment by oknet** [comment]
@zwoop please backport to 7.x, thanks

**Comment by oknet** [comment]
@zwoop Do you forget to backport it?

**Pull request merged**